src/Card.js

```diff
@@ -0,0 +1,126 @@
+function Card() {
+  this.frames = []
+  this.bonuses = [0, 0]
+};
+
+// at the moment frame.bowls are stored in frames array
+// refactoring to store full frame instance should clean up card logic
+// and return us to SRC principle
+
+Card.prototype.store = function(frame) {
+  if (!this._isEnd()) {
+    this.frames.push(frame.bowls)
+    // apply bonuses for the last frame before storing bonus for current frame
+    this._applyBonuses()
+    this._trackBonus(frame)
+  } else {
+    this._endGame()
+  }
+};
+
+Card.prototype._trackBonus = function(frame) {
+  if (frame.spare) {
+    this.bonuses[1] = 1
+  } else if (frame.strike) {
+    this.bonuses[1] = 2
+  }
+};
+
+Card.prototype._applyBonuses = function() {
+
+  // adds first bowl of recent round to frame two before,
+  //   only happens when two strikes occur in a row
+  if (this.bonuses[0]) {
+    this._addBonusRoll(this._twoPreviousFrame(), this._firstBowl())
+    this.bonuses[0] = 0
+  }
+
+  // if bonus is to be added to previous frame
+  if (this.bonuses[1]) {
+    // add first roll of current frame
+    this._addBonusRoll(this._previousFrame(), this._firstBowl())
+    this.bonuses[1] = 0
+
+    // if a second roll needs to be added
+    if (this._isPreviousFrameStrike()) {
+      // and the current frame has two bowls
+      if (!this._isCurrentFrameStrike()) {
+        // add second roll
+        this._addBonusRoll(this._previousFrame(), this._secondBowl())
+        this.bonuses[1] = 0
+      } else {
+        // store reminder to add second roll
+        this.bonuses[0] = 1
+      };
+    };
+  };
+};
+
+Card.prototype._isEnd = function() {
+  return (this.frames.length > 9 && this._isNoBonuses())
+};
+
+Card.prototype._isNoBonuses = function() {
+  return (this.bonuses[0] === this.bonuses[1])
+};
+
+// this function removes frames rolled for final frame bonus
+// it's necessary because the final frame logic doesn't exist
+Card.prototype._endGame = function() {
+  while (this.frames.length > 10) {
+    this.frames.pop()
+  };
+};
+
+Card.prototype.score = function() {
+  var score = 0
```

Comment:
This method really confuses me. What is the purpose of having a local array called frames when you already have a property this.frames? At the very least, the naming needs to be less ambiguous so that the purpose of each is clearer.